src/Cli/dotnet/Commands/Test/Terminal/TerminalTestReporter.cs

```diff
@@ -493,20 +478,17 @@ internal void TestCompleted(
             case TestOutcome.Timeout:
             case TestOutcome.Canceled:
             case TestOutcome.Fail:
-                asm.FailedTests++;
-                asm.TotalTests++;
```

Comment:
This existing code was not thread-safe already, and so my code in this PR is still not considering thread-safety.

@mariam-abdulla @nohwnd Can you please confirm if we can be concurrently processing two test result messages? In which case we have a bug here?

Reply:
Discussed offline with Mariam. Concurrency of test result messages can happen across different assemblies, but not from the same assembly. Since each assembly deals with its own `TestProgressState`, we are fine here.